File: app/src/main/java/com/raju/javabaseproject/ui/activities/base/BaseDaggerActivity.java
package com.raju.javabaseproject.ui.activities.base;

import com.raju.javabaseproject.viewmodels.base.BaseViewModel;

import javax.inject.Inject;

public class BaseDaggerActivity<V extends BaseViewModel> extends BaseActivity {

    @Inject
    V viewModel;

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if(viewModel != null) {
            viewModel.destroy();
        }
    }
}

File: app/src/main/java/com/raju/javabaseproject/ui/activities/base/BaseDaggerActivity.kt
package com.raju.javabaseproject.ui.activities.base

import com.raju.javabaseproject.viewmodels.base.BaseViewModel

import javax.inject.Inject

class BaseDaggerActivity<V : BaseViewModel> : BaseActivity() {

    @Inject
    lateinit var viewModel: V

    override fun onDestroy() {
        super.onDestroy()
        if (viewModel != null) {
            viewModel!!.destroy()
        }
    }
}
